spec/protocol.coffee

```diff
@@ -331,14 +351,34 @@ describe 'FBP runtime protocol', () ->
         chai.expect(payload).to.be.an 'object'
         chai.expect(payload).to.include.keys ['id', 'graph', 'component']
         chai.expect(payload.component).to.equal componentName
+        chai.expect(payload.metadata).to.eql add.metadata
         ui.removeListener 'message', checkMessage
         done()
       ui.on 'message', checkMessage
       add =
         id: 'mycoffeescriptproducer'
         graph: 'default/main'
         component: componentName
+        metadata:
+          label: 'myproducer'
       ui.send 'graph', 'addnode', add
+    it 'should be possible to change node metadata', (done) ->
+      checkMessage = (d, protocol, command, payload) ->
+        chai.expect(command, JSON.stringify(payload)).to.equal 'changenode'
+        chai.expect(protocol).to.equal 'graph'
+        chai.expect(payload).to.be.an 'object'
+        chai.expect(payload).to.include.keys ['id', 'graph', 'metadata']
+        chai.expect(payload.metadata).to.eql change.metadata
+        ui.removeListener 'message', checkMessage
+        done()
+      ui.on 'message', checkMessage
+      change =
+        id: 'mycoffeescriptproducer'
+        graph: 'default/main'
+        metadata:
+          label: 'mycoffeeproducer'
+          x: 2
+      ui.send 'graph', 'changenode', change
 
   describe 'subscribing to edges', ->
```

Comment:
Need a test for whether main graph, as returned by `getsource`, contains the right metadata. That is what Flowhub live-mode will mostly rely on